modules/ml/src/main/java/org/apache/ignite/ml/selection/paramgrid/ParamGrid.java

```diff
@@ -26,32 +30,129 @@
     /** Parameter values by parameter index. */
     private Map<Integer, Double[]> paramValuesByParamIdx = new HashMap<>();
 
+    /** Parameter names by parameter index. */
+    private Map<Integer, DoubleConsumer> settersByParamIdx = new HashMap<>();
+
     /** Parameter names by parameter index. */
     private Map<Integer, String> paramNamesByParamIdx = new HashMap<>();
 
     /** Parameter counter. */
     private int paramCntr;
 
+    /** Parameter search strategy. */
+    private HyperParameterSearchingStrategy paramSearchStgy = HyperParameterSearchingStrategy.BRUT_FORCE;
+
+    /** Satisfactory fitness to stop the hyperparameter search. */
+    private double satisfactoryFitness = 0.5;
+
+    /** Max tries to stop the hyperparameter search. */
+    private int maxTries = 100;
+
+    /** Seed. */
+    private long seed = 1234L;
+
     /** */
     public Map<Integer, Double[]> getParamValuesByParamIdx() {
-        return paramValuesByParamIdx;
+        return Collections.unmodifiableMap(paramValuesByParamIdx);
     }
 
     /**
      * Adds a grid for the specific hyper parameter.
      * @param paramName The parameter name.
+     * @param setter The method reference to trainer or preprocessor trainer setter.
      * @param params The array of the given hyper parameter values.
      * @return The updated ParamGrid.
      */
-    public ParamGrid addHyperParam(String paramName, Double[] params) {
+    public ParamGrid addHyperParam(String paramName, DoubleConsumer setter, Double[] params) {
         paramValuesByParamIdx.put(paramCntr, params);
         paramNamesByParamIdx.put(paramCntr, paramName);
+        settersByParamIdx.put(paramCntr, setter);
         paramCntr++;
         return this;
     }
 
+    /**
+     * Set up the hyperparameter searching strategy.
+     *
+     * @param paramSearchStgy Parameter search strategy.
+     */
+    public ParamGrid withParameterSearchStrategy(HyperParameterSearchingStrategy paramSearchStgy) {
+        this.paramSearchStgy = paramSearchStgy;
+        return this;
+    }
+
+    /**
+     *
```

Comment:
Javadoc in this class is missed in a lot of places.

Reply:
I suppose that you can use 'mvn initialize -Pjavadoc' for javadocs checks. It should complete without errors and warnings in our modules